### PR TITLE
fullName filling out

### PR DESCRIPTION
**Pull request by SofiaPetinova**
I created a new pull request, because prev. wasn't updated with current changes and I just messed up everything. I haven't closed prev. PR yet, I just thought maybe you want to see the past comments.

### CODE REVIEW

File: src/components/firstProfile/NameProfile.tsx [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+  Button,
+  FormControl,
+  FormHelperText,
+  OutlinedInput,
+  Typography,
+} from '@mui/material'
+import { useState } from 'react'
+import useProfileData from '../../hooks/useProfileData'
+import axios from 'axios'
+// import { v4 as uuidv4 } from 'uuid'
```
Comment (Svitlana-G) [inline src/components/firstProfile/NameProfile.tsx:13]: Please delete comments

File: src/components/firstProfile/NameProfile.tsx [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    ''
+  )
+  const [error, setError] = useState<string>('')
+  const [isValid, setIsValid] = useState<boolean>(false)
+
+  // const userId = //
+  // const requestBody = {
+  //   id: userId,
+  //   name: fullName,
+  // }
```
Comment (Svitlana-G) [inline src/components/firstProfile/NameProfile.tsx:31]: Please delete comments

File: src/components/firstProfile/NameProfile.tsx [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+  const handleSubmit = async (e: { preventDefault: () => void }) => {
+    e.preventDefault()
+
+    if (fullName.trim() === '') {
+      setError('Please enter your full name')
+    } else if (!FULLNAME_REGEX.test(fullName)) {
+      setIsValid(true)
+    } else {
+      try {
+        await axios.post('api/addName', fullName)
```
Comment (Svitlana-G) [inline src/components/firstProfile/NameProfile.tsx:42]: I think we save the data at the end when all the fields in the profile are filled in. I propose to clarify this point

File: src/routes/index.tsx [hunk 4]
```diff
@@ -119,6 +120,10 @@ const routes: RouteObject[] = [
       },
     ],
   },
+  {
+    path: 'name',
+    element: <NameProfile />,
+  },
```
Comment (Svitlana-G) [inline src/routes/index.tsx:126]: I think it should be as a child in the  'user' path

File: src/styles/firstProfileStyles.tsx [hunk 5]
```diff
@@ -0,0 +1,65 @@
+import { makeStyles } from 'tss-react/mui'
+export const firstProfileStyles = makeStyles()((theme) => {
+  return {
+    mainBox: {
```
Comment (Svitlana-G) [inline src/styles/firstProfileStyles.tsx:4]: Please use commonStyles

File: src/styles/firstProfileStyles.tsx [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+      marginLeft: 20,
+      marginRight: 20,
+      [theme.breakpoints.up('sm')]: {
+        width: 400,
+        margin: '0 auto',
+      },
+    },
+    title: {
```
Comment (Svitlana-G) [inline src/styles/firstProfileStyles.tsx:12]: Please use commonStyles

File: src/styles/firstProfileStyles.tsx [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+    },
+    profileInput: {
+      backgroundColor: '#FFF1EC',
+      borderRadius: 10,
+      outline: 'none',
+      '&.MuiOutlinedInput-root .MuiOutlinedInput-notchedOutline': {
+        border: 'none',
+      },
+    },
+    nextButton: {
```
Comment (Svitlana-G) [inline src/styles/firstProfileStyles.tsx:34]: It would be good to put it in the commonStyles

File: src/components/firstProfile/NameProfile.tsx [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
```
Comment (LyubovSugak) [inline src/components/firstProfile/NameProfile.tsx:35]: api calls should be moved to separate function/file in the folder action. It's a principle of splitting up app calls, business logic and presentational components. My advice to read about above principle.
Anyway as Svetlana mentioned we use local storage to store data first there

File: src/components/firstProfile/NameProfile.tsx [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+  }
+
+  return (
+    <Box className={classes.mainBox}>
+      <Logo />
+      <Typography variant="h1" className={classes.title} pt={10}>
+        Let&apos;s get started!
+      </Typography>
+      <Typography variant="body1" className={classes.profileText} pt={5}>
+        What&apos;s your name?
```
Comment (LyubovSugak) [inline src/components/firstProfile/NameProfile.tsx:60]: Typo?

File: src/hooks/useProfileData.ts [hunk 10]
```diff
@@ -0,0 +1,16 @@
+import { useState, useEffect } from 'react'
+
+const useProfileData = <T>(key: string, initialValue: T) => {
```
Comment (LyubovSugak) [inline src/hooks/useProfileData.ts:3]: Nice idea to use hook, but could you refactor it to make generic and use in other components. Plz thing about naming (in useState),change to smth meaningful

File: src/components/firstProfile/NameProfile.tsx [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+    'profileData.fullName',
+    ''
+  )
+  const [error, setError] = useState<string>('')
+  const [isValid, setIsValid] = useState<boolean>(false)
+
+  const handleSubmit = async (e: { preventDefault: () => void }) => {
+    e.preventDefault()
+
+    if (fullName.trim() === '') {
```
Comment (LyubovSugak) [inline src/components/firstProfile/NameProfile.tsx:29]: Plz refactor lines underneath, I don't think you need spaghetti here

File: src/components/firstProfile/NameProfile.tsx (hunk 9 above)
Comment (SofiaPetinova) [reply src/components/firstProfile/NameProfile.tsx:60]: нет, когда я ставлю обычный ', то выходит ошибка то ли eslint, то ли prettier. Погуглила и предлагается такой вариант.

File: src/components/firstProfile/NameProfile.tsx (hunk 8 above)
Comment (SofiaPetinova) [reply src/components/firstProfile/NameProfile.tsx:35]: api calls удаляю тогда вообще?

File: src/hooks/useProfileData.ts (hunk 10 above)
Comment (SofiaPetinova) [reply src/hooks/useProfileData.ts:3]: это вообще удаляю? тк ты мне написала оставить только верстку и логику кнопка

File: src/components/firstProfile/NameProfile.tsx [hunk 12]
```diff
@@ -0,0 +1,134 @@
...
+        Let&apos;s get started!
+      </Typography>
+      <Typography
+        variant="body1"
+        className={`${commonClasses.text} ${classes.profileText}`}
+        pt={5}
+      >
+        What&apos;s your name?
+      </Typography>
+      <form onSubmit={handleSubmit}>
```
Comment (LyubovSugak) [inline src/components/firstProfile/NameProfile.tsx:55]: don't need `form` here. Clicking on the button will add data to local storage. The form usually is needed to send data via API

File: src/components/firstProfile/NameProfile.tsx [hunk 13]
```diff
@@ -0,0 +1,134 @@
...
+          ></OutlinedInput>
+          {isValid && (
+            <FormHelperText sx={{ color: '#1D878C' }}>
+              15 symbols max.
+            </FormHelperText>
+          )}
+          {error && (
+            <FormHelperText sx={{ color: '#1D878C' }}>{error}</FormHelperText>
+          )}
+          <Button
```
Comment (LyubovSugak) [inline src/components/firstProfile/NameProfile.tsx:71]: you can remove button, component `Name profile`will be shown via generic component carousel

File: src/components/firstProfile/NameProfile.tsx [hunk 14]
```diff
@@ -0,0 +1,134 @@
...
+            fullWidth
+            variant="outlined"
+            className={`${commonClasses.submitButton} ${classes.nextButton}`}
+            type="submit"
+            onClick={handleSubmit}
+          >
+            Next
+          </Button>
+          <Typography variant="h2" className={classes.dot}>
```
Comment (LyubovSugak) [inline src/components/firstProfile/NameProfile.tsx:80]: also plz remove dots

File: src/components/firstProfile/NameProfile.tsx [hunk 15]
```diff
@@ -0,0 +1,134 @@
...
+            <span className={classes.span}>.</span>....
+          </Typography>
+        </FormControl>
+      </form>
+    </Box>
+  )
+}
+export default NameProfile
+
+const useStyles = makeStyles()(() => {
```
Comment (LyubovSugak) [inline src/components/firstProfile/NameProfile.tsx:90]: After refactoring, remove plz unused code in style hook

File: src/routes/index.tsx [hunk 16]
```diff
@@ -63,6 +64,10 @@ const routes: RouteObject[] = [
   {
     path: 'user',
     children: [
+      {
```
Comment (LyubovSugak) [inline src/routes/index.tsx:67]: plz remove after the route, there will be only one route for the whole Profile

**Review by LyubovSugak — CHANGES_REQUESTED** [review]

**Pull request closed without merging**